### DISCUSSION TITLE
Emojis don't show up on RHEL 7

### DISCUSSION THREAD

**Discussion by halloleo** [starship/starship#5738] [Q&A] [answered]
On a RHEL 7 system I don't get the emojis to work. 

I went through [your suggestions for this case](https://starship.rs/faq/#why-don-t-i-see-a-glyph-symbol-in-my-prompt) and I have:
* Set locale to UTF-8 (and `LC_ALL` is `en_US.UTF-8`).
* Installed the Google Noto fonts (via `yum install google-noto-emoji-color-fonts.noarch` and `yum install google-noto-emoji-fonts.noarch`)
* Installed and in my terminal switched to the Nerd font `DejaVuSansMNerdFontMono-Regular`.

Now, when I issue `echo -e "\xf0\x9f\x90\x8d"` I get garbage, ***not*** the snake. But, when I execute `echo -e "\xee\x82\xa0"`, I ***do*** get the branch symbol.

So it looks like the Emojis from Google Noto are not reached. What can I do? Any ideas?

**Comment by chipbuster** [comment]
It's possible that font fallback isn't automatically configured on RHEL, though I haven't used anything RedHat in a while now. What terminal emulator are you using, and have you configured font fallback manually?

**Reply by chipbuster** [reply]
For reference, [here's how I did font fallback when I was initially doing this stuff](https://github.com/starship/starship/discussions/1600#discussioncomment-52271):

In `~/.config/fontconfig/fonts.conf`, add:

```
    <match target="pattern">
        <test qual="any" name="family"><string>monospace</string></test>
        <edit name="family" mode="append" binding="weak"><string>EmojiOne</string></edit>
    </match>
    <match target="pattern">
        <test qual="any" name="family"><string>serif</string></test>
        <edit name="family" mode="append" binding="weak"><string>EmojiOne</string></edit>
    </match>
    <match target="pattern">
        <test qual="any" name="family"><string>sans-serif</string></test>
        <edit name="family" mode="append" binding="weak"><string>EmojiOne</string></edit>
    </match>
    <match target="pattern">
        <test qual="any" name="family"><string>emoji</string></test>
        <edit name="family" mode="append" binding="weak"><string>EmojiOne</string></edit>
    </match>
```

Then run `fc-cache -f -v` and restart the terminal. You'll probably need to change the font name (I was using EmojiOne here) and you'd need to make sure you have `monospace` selected as a fallback font in your terminal emulator, since this configuration only sets the fallback for the `monospace`, `serif`, and `sans-serif` fonts.

**Reply by halloleo** [reply]
Thanks heaps for chiming in @chipbuster! 

I'm using vterm in Emacs and the build-in terminal emulator in the (Windows) X server MobaXterm.

I didn't have any user `fonts.conf` file, so I created one containing your snippet:

```
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
<!-- $XDG_CONFIG_HOME/fontconfig/fonts.conf for per-user font configuration -->
<fontconfig>
    <match target="pattern">
        <test qual="any" name="family"><string>monospace</string></test>
        <edit name="family" mode="append" binding="weak"><string>NotoEmoji</string></edit>
    </match>
    <match target="pattern">
        <test qual="any" name="family"><string>serif</string></test>
        <edit name="family" mode="append" binding="weak"><string>NotoEmoji</string></edit>
    </match>
    <match target="pattern">
        <test qual="any" name="family"><string>sans-serif</string></test>
        <edit name="family" mode="append" binding="weak"><string>NotoEmoji</string></edit>
    </match>
    <match target="pattern">
        <test qual="any" name="family"><string>emoji</string></test>
        <edit name="family" mode="append" binding="weak"><string>NotoEmoji</string></edit>
    </match>
</fontconfig>
```

I'm using NotoEmoj as emoji font: `fc-list` lists it as 

    /usr/share/fonts/google-noto-emoji/NotoEmoji-Regular.ttf: Noto Emoji:style=Regular

But the snake still does not appear. :(

My `fonts.conf` is certainly read by fc-cache, my `$XDG_CONFIG_HOME` is set to the standard `/home/myuser/.config`.

Do I need to copy other stuff into the `fonts.conf`?

**Reply by chipbuster** [reply, marked as answer]
Oh, I'm sorry, I totally misunderstood your setup.

Outside of very specific circumstances (basically involving hyperspecialized terminal emulators like Warp), the display of fonts is handled by the system that does the displaying. So in this case, your real problem is likely that your terminal emulator and host system pair (Windows and MobaXTerm) aren't supporting Unicode emojis.

I found [a thread](https://github.com/romkatv/powerlevel10k/issues/674) on pl10k which dealt with this issue, but the primary finding seems to be that the MobaXTerm folks are aware of it but haven't solved it yet (?) and that most people who have experienced this issue have moved to something like Windows Terminal.

**Reply by halloleo** [reply]
Interesting! 

But I install the fonts on the remote machine and can only use these fonts in Emacs and not fonts installed on the displaying local machine (the X "server"). -- I guess this is maybe more an emacs/vterm problem than anything else...

Anyway, thank you so much for trying to help! I really appreciate it!
